File: replace_v2.py
import os
import re

directories = ['src/app', 'src/components']

heading_regex = re.compile(r'(<h[1-6][^>]*className=["\'][^"\']*)(text-foreground)([^"\']*["\'][^>]*>)')
bold_regex = re.compile(r'(<h[1-6][^>]*className=["\'][^"\']*)(font-semibold|font-bold)([^"\']*["\'][^>]*>)')

def process_file(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    original_content = content
    
    # 1. Update Headings: text-foreground -> text-primary dark:text-white
    while True:
        new_content = heading_regex.sub(r'\1text-primary dark:text-white\3', content)
        if new_content == content:
            break
        content = new_content
        
    # 2. Update Headings: font-bold/semibold -> font-extrabold tracking-tight
    while True:
        new_content = bold_regex.sub(r'\1font-extrabold tracking-tight\3', content)
        if new_content == content:
            break
        content = new_content

    # 3. Gold -> Accent Replacements globally
    content = re.sub(r'\bgradient-text-gold\b', 'text-accent', content)
    content = re.sub(r'\btext-gold\b', 'text-accent', content)
    content = re.sub(r'\bbg-gold\b', 'bg-accent', content)
    content = re.sub(r'\bborder-gold\b', 'border-accent', content)
    
    # 4. Paragraph readability upgrade
    # From: text-sm text-muted-foreground leading-relaxed
    # To: text-base text-muted-foreground leading-loose
    content = content.replace('text-sm text-muted-foreground leading-relaxed', 'text-base text-muted-foreground leading-loose')
    
    if content != original_content:
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(content)
        print(f'Updated {filepath}')

for d in directories:
    for root, _, files in os.walk(d):
        for file in files:
            if file.endswith('.tsx'):
                process_file(os.path.join(root, file))

File: replace_v2.js
const fs = require('fs');
const path = require('path');

const directories = ['src/app', 'src/components'];

const headingRegex = /(<h[1-6][^>]*className=["'][^"']*?)(text-foreground)([^"']*?["'][^>]*>)/g;
const boldRegex = /(<h[1-6][^>]*className=["'][^"']*?)(font-semibold|font-bold)([^"']*?["'][^>]*>)/g;

function processFile(filepath) {
    let content = fs.readFileSync(filepath, 'utf-8');
    const originalContent = content;
    
    // 1. Update Headings: text-foreground -> text-primary dark:text-white
    let newContent;
    while (true) {
        newContent = content.replace(headingRegex, '$1text-primary dark:text-white$3');
        if (newContent === content) break;
        content = newContent;
    }
        
    // 2. Update Headings: font-bold/semibold -> font-extrabold tracking-tight
    while (true) {
        newContent = content.replace(boldRegex, '$1font-extrabold tracking-tight$3');
        if (newContent === content) break;
        content = newContent;
    }

    // 3. Gold -> Accent Replacements globally
    content = content.replace(/\bgradient-text-gold\b/g, 'text-accent');
    content = content.replace(/\btext-gold\b/g, 'text-accent');
    content = content.replace(/\bbg-gold\b/g, 'bg-accent');
    content = content.replace(/\bborder-gold\b/g, 'border-accent');
    content = content.replace(/\bbg-gold\//g, 'bg-accent/');
    content = content.replace(/\bborder-gold\//g, 'border-accent/');
    
    // 4. Paragraph readability upgrade
    content = content.replace(/text-sm text-muted-foreground leading-relaxed/g, 'text-base text-muted-foreground leading-loose');
    
    if (content !== originalContent) {
        fs.writeFileSync(filepath, content, 'utf-8');
        console.log(`Updated ${filepath}`);
    }
}

function walkDir(dir) {
    const files = fs.readdirSync(dir);
    for (const file of files) {
        const filepath = path.join(dir, file);
        const stat = fs.statSync(filepath);
        if (stat.isDirectory()) {
            walkDir(filepath);
        } else if (filepath.endsWith('.tsx')) {
            processFile(filepath);
        }
    }
}

directories.forEach(d => {
    if (fs.existsSync(d)) {
        walkDir(d);
    }
});
